File: codeforces/1324/B/1324B.java
import java.io.*;


public class Main {
public static void main(String [] argv) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String cantidad = br.readLine();
        String [] solucion= new String[Integer.parseInt(cantidad)];
        for(int i=0;i<Integer.parseInt(cantidad);i++){
            String cantidadfor = br.readLine();
            int cantidadelementos=Integer.parseInt(cantidadfor);
            String [] elementosS =br.readLine().split(" ");
            int[] elementos=new int[cantidadelementos];
            for(int j=0;j<elementosS.length;j++){
                elementos[j]=Integer.parseInt(elementosS[j]);
            }
            for(int j=0;j<cantidadelementos;j++){
                boolean centinela=false;
                for(int k=cantidadelementos-1;k>=0;k--){
                    if(elementos[j]==elementos[k] && k-j>=2){
                        solucion[i]="YES";
                        break;
                    } 
                }
                if(centinela) break;
                
            }
        
                
        }
        for(int i=0;i<solucion.length;i++){
            if(solucion[i]==null){
                solucion[i]="NO";
            }
        }
        for(int i=0;i<solucion.length;i++){
        System.out.println(solucion[i]);
        }
    }
}

File: codeforces/1324/B/1324B.kt
package codeforces

import java.io.BufferedReader
import java.io.InputStreamReader

tailrec fun palindrome(head : String , tail : List<String>) : Boolean {
    return if(tail.isEmpty())
        return false;
    else if(tail.first() == head)
        true;
    else
        palindrome(head, tail.drop(1))
}

fun main() {
    val In = BufferedReader(InputStreamReader(System.`in`))
    val t = In.readLine().toInt()
    for( i in 1..t) {
       In.readLine()
       var res = "NO"
       val s = In.readLine().split(" ").toList()
       for(c in s){
           if(palindrome(c, s.drop(2)))
               res = "YES"
               break
       }
       println(res)
    }
}